synapse/handlers/presence.py

```diff
@@ -659,27 +659,28 @@ def __init__(self, hs: "HomeServer"):
         )
 
         now = self.clock.time_msec()
-        for state in self.user_to_current_state.values():
-            self.wheel_timer.insert(
-                now=now, obj=state.user_id, then=state.last_active_ts + IDLE_TIMER
-            )
-            self.wheel_timer.insert(
-                now=now,
-                obj=state.user_id,
-                then=state.last_user_sync_ts + SYNC_ONLINE_TIMEOUT,
-            )
-            if self.is_mine_id(state.user_id):
+        if self._presence_enabled:
```

Comment:
Feels a bit weird, Would it maybe make more sense to not register the prescence handler if presence is disabled? :shrug:

Reply:
Yeah, though note we do `self._presence_enabled` checks in a lot of places already. We could change the way we handle presence being disabled (and I agree that it'd quite probably be nicer to just never have a `PresenceHandler`), but that is a refactor I'd rather not take on here.

Reply:
extremely reasonable, perfectly proper, utterly understandable, I can't think of more synonyms for "very fair"